### PR TITLE
Implement Wt signals

### PR DESCRIPTION
**Pull request by zer0main**

### CODE REVIEW

File: src/luawt/globals.hpp [hunk 1]
```diff
@@ -10,24 +10,28 @@
...
 #include <boost/cast.hpp>
+#include <boost/shared_ptr.hpp>
 
 #include "boost-xtime.hpp"
 #include <Wt/WApplication>
 #include <Wt/WContainerWidget>
 #include <Wt/WEnvironment>
 #include <Wt/WServer>
 #include <Wt/WText>
+#include <Wt/WObject>
```
Comment (starius) [inline src/luawt/globals.hpp]: sort includes, please

File: src/luawt/globals.hpp [hunk 2]
```diff
@@ -227,6 +244,101 @@ struct wrap {
...
+    }
+
+    ~SlotWrapper() {
+        if (app_->L()) {
+            luaL_unref(app_->L(), LUA_REGISTRYINDEX, func_id_);
+        }
+    }
+
+    int func_id_;
+    luawt_Application* app_;
```
Comment (starius) [inline src/luawt/globals.hpp:265]: Add a comment why to store a pointer to `app_` here, please.

**Review by starius — CHANGES_REQUESTED** [review]

File: src/luawt/globals.hpp (hunk 1 above)
Comment (zer0main) [reply src/luawt/globals.hpp]: Done

File: src/luawt/globals.hpp (hunk 2 above)
Comment (zer0main) [reply src/luawt/globals.hpp:265]: Done

**Review by starius — APPROVED** [review]

File: src/luawt/globals.hpp (hunk 2 above)
Comment (starius) [reply src/luawt/globals.hpp:265]: LGTM

File: src/luawt/globals.hpp (hunk 1 above)
Comment (starius) [reply src/luawt/globals.hpp]: LGTM

**Pull request merged**